api.js

```diff
@@ -1536,6 +1537,65 @@ API.prototype.getTraceMetadata = function getTraceMetadata() {
   return metadata
 }
 
+/**
+ * Get a set of tracked identifiers
+ *
+ * @param {object} params Input parameters.
+ * @param {string} params.responseId The LLM generated identifier for the
+ * response.
+ * @returns {LlmTrackedIds|undefined} The tracked identifiers.
+ */
+API.prototype.getLlmTrackedIds = function getLlmMessageIds({ responseId } = {}) {
+  const tx = this.agent.tracer.getTransaction()
+  if (!tx) {
+    return logger.warn("No transaction found for response_id '%s'.", responseId)
+  }
+  return tx.llm.conversation_id_tracker.get(responseId)
+}
+
+/**
+ * Generate a new LLM feedback custom event and ship it to the tracked events.
+ *
+ * @param {object} params Input parameters.
+ * @param {string} [params.conversationId=""] If available, the unique
+ * identifier for the LLM conversation that triggered the event.
+ * @param {string} [params.requestId=""] If available, the request identifier
+ * from the remote service.
+ * @param {string} params.messageId Identifier for the message being rated.
+ * Obtained from {@link getLlmTrackedIds}.
+ * @param {string} params.category A tag for the event.
+ * @param {string} params.rating A indicator of how useful the message was.
+ * @param {string} [params.message=""] The message that triggered the event.
+ * @param {object} [params.metadata={}] Additional key-value pairs to associate
+ * with the recorded event.
+ */
+API.prototype.recordLlmFeedbackEvent = function recordLlmFeedbackEvent({
+  conversationId = '',
+  requestId = '',
+  messageId,
+  category,
+  rating,
+  message = '',
+  metadata = {}
+} = {}) {
+  const tx = this.agent.tracer.getTransaction()
+  if (!tx) {
+    return logger.warn(
+      'No message feedback events will be recorded. record_llm_feedback_event must be called within the scope of a transaction.'
+    )
+  }
+
+  const feedback = new LlmFeedbackMessage({
```

Comment:
ah crap, i messed up LlmFeedbackMessage.  It should use the metadata passed in and if not defined it should also check agent.llm.metadata otherwise it'd be empty